src/posit/connect/external/databricks.py

```diff
@@ -160,277 +91,401 @@ def __call__(self) -> Dict[str, str]:
+    from shiny import reactive
+    from shiny.express import render, session
 
-    DATABRICKS_HOST = "<REDACTED>"
-    DATABRICKS_HOST_URL = f"https://{DATABRICKS_HOST}"
-    SQL_HTTP_PATH = "<REDACTED>"
+    from posit.connect.external.databricks import ConnectStrategy
 
-    # NOTE: currently the databricks_cli local strategy only supports auth code OAuth flows.
-    # https://github.com/databricks/cli/issues/1939
-    #
-    # This means that the databricks_cli supports local development using the developer's
-    # databricks credentials, but not the credentials for a service principal.
-    # To fallback to service principal credentials in local development, use
-    # `PositLocalContentCredentialsStrategy` as a drop-in replacement.
-    posit_strategy = PositContentCredentialsStrategy(local_strategy=databricks_cli)
 
-    cfg = Config(host=DATABRICKS_HOST_URL, credentials_strategy=posit_strategy)
+    @reactive.calc
+    def cfg():
+        session_token = session.http_conn.headers.get("Posit-Connect-User-Session-Token")
+        return Config(
+            credentials_strategy=ConnectStrategy(user_session_token=session_token),
+            host=os.getenv("DATABRICKS_HOST"),
+        )
 
-    databricks_user_info = CurrentUserAPI(ApiClient(cfg)).me()
-    print(f"Hello, {databricks_user_info.display_name}!")
 
-    query = "SELECT * FROM samples.nyctaxi.trips LIMIT 10;"
-    with sql.connect(
-        server_hostname=DATABRICKS_HOST,
-        http_path=SQL_HTTP_PATH,
-        credentials_provider=posit_strategy.sql_credentials_provider(cfg),
-    ) as connection:
-        with connection.cursor() as cursor:
-            cursor.execute(query)
-            rows = cursor.fetchall()
-            print(pd.DataFrame([row.asDict() for row in rows]))
+    @render.text
+    def text():
+        databricks_user_info = CurrentUserAPI(ApiClient(cfg())).me()
+        return f"Hello, {databricks_user_info.display_name}!"
+    ```
+
+    This example shows how to do Service Account authentication with a Service Account Databricks OAuth integration.
+
+    ```python
+    import os
+
+    from databricks.sdk.core import ApiClient, Config
+    from databricks.sdk.service.iam import CurrentUserAPI
+    from shiny import reactive
+    from shiny.express import render
+
+    from posit.connect.external.databricks import ConnectStrategy
+
+
+    @reactive.calc
+    def cfg():
+        return Config(
+            credentials_strategy=ConnectStrategy(),
+            host=os.getenv("DATABRICKS_HOST"),
+        )
+
+
+    @render.text
+    def text():
+        databricks_user_info = CurrentUserAPI(ApiClient(cfg())).me()
+        return f"Hello, {databricks_user_info.display_name}!"
     ```
     """
 
     def __init__(
         self,
-        local_strategy: CredentialsStrategy,
         client: Optional[Client] = None,
+        user_session_token: Optional[str] = None,
     ):
-        self._local_strategy = local_strategy
+        self._cp = None
```

Comment:
nit; add a type signature for clarity on the non-null state.

```suggestion
        self._cp: CredentialsProvider | None = None
```